pkg/pravega/pravega_segmentstore.go

```diff
@@ -245,3 +324,70 @@ func configureTier2Filesystem(podSpec *corev1.PodSpec, pravegaSpec *api.PravegaS
 		})
 	}
 }
+
+func makeSegmentStoreHeadlessService(pravegaCluster *api.PravegaCluster) *corev1.Service {
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      k8sutil.HeadlessServiceNameForSegmentStore(pravegaCluster.Name),
+			Namespace: pravegaCluster.Namespace,
+			Labels:    k8sutil.LabelsForSegmentStore(pravegaCluster),
+			OwnerReferences: []metav1.OwnerReference{
+				*k8sutil.AsOwnerRef(pravegaCluster),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "server",
+					Port:     12345,
```

Comment:
Where does 12345 come from?  Seems to me that the normal segment store port should be used here.   Meanwhile, in a NodePort situation, K8s supplies the port AFAIK.

Reply:
Isn't `12345` the default port for the Segment Store?

Reply:
https://github.com/pravega/pravega/blob/master/config/config.properties#L55

Reply:
Hmm did not know that.   Wow does that suck!